### PR TITLE
Added the ability to fall back to another image (fixes #33)

### LINKED ISSUE #33: Adding the ability to fall back to another image
It would be awesome if an image could fall back to another if it fails or is not specified.  The changes necessary for this would not be too difficult.

### PR DESCRIPTION
**Pull request by damoninfusion**
Added the ability to fall back to another image if the src image fails to load or is missing.

Added examples to demo page.

### CODE REVIEW

**Comment by damoninfusion** [comment]
I signed it!

File: iron-image.html [hunk 1]
```diff
@@ -178,6 +183,22 @@
...
+       */
+      fallback: {
+        type: String,
+        value: null
+      },
+
+      /**
+       * Did the src image fail to load?
+       */
+      srcFail: {
```
Comment (valdrinkoshi) [inline iron-image.html]: this property should be read-only

File: iron-image.html [hunk 2]
```diff
@@ -361,6 +390,17 @@
       } else {
         this.$.img.src = this.src || '';
       }
+    },
+
+    _fallbackCheck: function() {
```
Comment (valdrinkoshi) [inline iron-image.html]: this is a potential forever-loop, in case `fallback` also fails!

File: iron-image.html [hunk 3]
```diff
@@ -361,6 +390,17 @@
...
+      if (this.fallback) {
+        img = new Image();
+        img.src = this.src;
+        img.onerror = function() {
+          this.srcFail = true;
+          this.src = this.fallback;
```
Comment (valdrinkoshi) [inline iron-image.html]: `src` shouldn't be overridden, we should only update `this.$.img.src`

**Comment by kevin-wise-infusionsoft** [comment]
I confirm

**Comment by damoninfusion** [comment]
I confirm

**Comment by damoninfusion** [comment]
I confirm consent @kevin-wise-infusionsoft

**Comment by kevin-wise-infusionsoft** [comment]
I confirm consent @damoninfusion

**Comment by opus1269** [comment]
I confirm consent

**Comment by opus1269** [comment]
I confirm consent @opus1269

**Comment by damoninfusion** [comment]
I confirm consent

**Comment by kevin-wise-infusionsoft** [comment]
I confirm consent

**Comment by damoninfusion** [comment]
I rebased this PR on master.  Can't seem to make the GoogleBot happy.  Not sure what's going on.

**Comment by kevin-wise-infusionsoft** [comment]
I rebased this PR on master.  Can't seem to make the GoogleBot happy.  Not sure what's going on.

**Comment by bicknellr** [comment]
FYI, #44 is coming in soon which should cover this use case: it's going to provide an error property that you can bind to (which also means you can listen for `error-changed`).

**Comment by damoninfusion** [comment]
@bicknellr I looked over #44 and it looks like it won't actually set a fallback image, it just sets an error state and it would be up to an external component to respond to that.  Is that correct?  If that is the case I think there is still a need for a fallback. I'd be happy to build that in on top of your branch if you see any value there.  Let me know what you think.

**Comment by kevin-wise-infusionsoft** [comment]
@bicknellr I looked over #44 and it looks like it won't actually set a fallback image, it just sets an error state and it would be up to an external component to respond to that.  Is that correct?  If that is the case I think there is still a need for a fallback. I'd be happy to build that in on top of your branch if you see any value there.  Let me know what you think.

**Comment by bicknellr** [comment]
@kevin-wise-infusionsoft That's correct, #44 does not provide a way to supply a fallback URL directly. I think this will be more flexible, even though it makes the common case of providing a known good URL as a backup slightly more complex. [A reusable wrapper that adds a fallback URL property using the error property](http://jsbin.com/punatijafi/1/edit?html,console,output) is pretty minimal though.

**Comment by kevin-wise-infusionsoft** [comment]
@bicknellr So would you consider a PR that adds a fallback URL?  I'm happy to create that but don't need to waste both of our time if you really don't want to go that direction.

**Comment by bicknellr** [comment]
@kevin-wise-infusionsoft I talked with @notwaldorf and @cdata about this just a little while ago and no, I don't think we want to add a fallback URL. :/ While it does make a reasonably typical problem a bit easier, a fallback URL doesn't seem robust enough on its own and it would be better to have only one way of handling errors.

p.s. [Here's an updated version of the wrapper](http://jsbin.com/xegisavito/edit?html,output) (still dependent upon #44) that doesn't require every iron-image to be nested in another element.

**Pull request closed without merging**